Social Authentication

pbteja1998 opened the pull request:
- Added **Github Social Login**
- Added **Linkedin Social Login**
- Started maintaining separate file **API.md** to document API end points
- Started maintaining **CHANGELOG.md** for every new pull request and list important features implemented
- Get and Edit **Auth User**
- Get and Edit **User Profile**
- Configure **CORS Headers** to allow only **coderplex.org**

rrmerugu commented:
@pbteja1998 the code looks clean, im not able to test it locally. Can you make write a change.log with the APIs and features you implemented. And I will merge this.

pbteja1998 commented:
@rrmerugu I have added the API end points that I implemented in the Readme, should I add a new file with change.log and write those changes in it too?  Why are you not able to test it locally? What might be the problem?

pbteja1998 commented on webapp/coderplex_apis/settings.py:
```diff
@@ -140,4 +158,23 @@
+EMAIL_PORT = 587
+EMAIL_USE_TLS = True
+EMAIL_HOST_USER = 'pbteja1998@gmail.com'
+EMAIL_HOST_PASSWORD = ''
```
@rrmerugu The above settings need to be changed, the email has to be some coderplex email, while testing I filled EMAIL_HOST_PASSWORD with my password and tested it,it works fine. So while hosting you have to create some env variables for these. I don't know how exactly to achieve that, so I left it blank.

rrmerugu replied:
hey, why don't you convert  the email and password into os.environ.get('EMAIL_HOST_USER') and os.environ.get('EMAIL_HOST_PASSWORD') . so we can take care of the details while deploying.

pbteja1998 commented on webapp/coderplex_apis/settings.py:
```diff
@@ -140,4 +158,23 @@
+SOCIALACCOUNT_EMAIL_VERIFICATION = 'optional'
+SOCIALACCOUNT_EMAIL_REQUIRED = False
+SOCIALACCOUNT_QUERY_EMAIL = True
+
```
I don't know what the above three settings indicate, so I set those with the same values as in the sample app you provided .

pbteja1998 commented on webapp/coderplex_apis/settings.py:
```diff
@@ -77,17 +89,23 @@
+        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
     }
 }
 
```
I replaced Postgres database with sqlite for convenience. Should I revert back to the way it was before?

rrmerugu replied:
can we revert this to the Postgres, we can make this config specific to localhost and production settings.  I mean, `localhost.py` uses sqlite3 and `production.py` uses Postgres later

pbteja1998 commented on webapp/authentication/views.py:
```diff
@@ -0,0 +1,14 @@
+class LinkedinLogin(SocialLoginView):
+    adapter_class = LinkedInOAuth2Adapter
+    client_class = OAuth2Client
+    callback_url = "http://127.0.0.1:8000/callback"
```
A new github and LinkedIn oauth 2.0 coderplex application need to be created with call back URLs as above and those credentials need to be added as social applications  in the Postgres database that we have been using.

pbteja1998 replied:
After it is hosted, will the base url be automatically resolved or should we change it explicitly with base url of this application in AWS?

rrmerugu replied:
if callback_url should be in the code, why don't you create it as `LINKEDIN_CALLBACK_URL= os.environ.get("LINKEDIN_CALLBACK_URL ", "https://127.0.0.1:8000/callback")` in `settings.py` , so in the production, we can set LINKEDIN_CALLBACK_URL and it will take that.

rrmerugu replied:
you should be able to access LINKEDIN_CALLBACK_URL from settings with `from django.conf import settings` and `settings. LINKEDIN_CALLBACK_URL`

pbteja1998 commented on webapp/authentication/serializers.py:
```diff
@@ -0,0 +1,7 @@
+    code = serializers.CharField(max_length=100)
+
+class LinkedInCodeSerializer(serializers.Serializer):
+    code = serializers.CharField(max_length=100)
```
I don't know exactly where we are using these serializers. I thought it is being used somewhere in the code of django-allauth, if not I will remove it.

rrmerugu commented:
@pbteja1998  its not about some code issue, im not to find some time to setup locally and test each component.

pbteja1998 commented on webapp/member/serializers.py:
```diff
@@ -1,8 +1,72 @@
+        instance.interested_technologies = validated_data.get('interested_technologies', instance.interested_technologies)
+        instance.user = user
+        instance.save()
+        return instance
```
Is the above implementation correct? Can the update function in serializer take 3 arguments?

rrmerugu replied:
use `ModelSerializer` instead of `Serializer`.

pbteja1998 commented:
@rrmerugu I made the requested changes. But User Profile GET, POST apis are not working. I do not know the problem. Can you check it? Other than that Everything is working fine.

pbteja1998 commented on webapp/coderplex_apis/config/production.py:
```diff
@@ -0,0 +1,24 @@
+DEBUG = False
+ALLOWED_HOSTS = [ 'coderplex.org', '*.coderplex.org']
+CORS_ORIGIN_WHITELIST = ['coderplex.org', '*.coderplex.org']
+
```
@rrmerugu Is this the correct way for allowing only coderplex?

pbteja1998 commented on webapp/coderplex_apis/config/localhost.py:
```diff
@@ -0,0 +1,20 @@
+DEBUG = True
+ALLOWED_HOSTS = ['*']
+CORS_ORIGIN_ALLOW_ALL = True
+
```
@rrmerugu Is this the correct way for allowing all hosts?
